config/validate.go

```diff
@@ -0,0 +1,89 @@
+package config
+
+import (
+	"errors"
+
+	"github.com/mrz1836/go-cachestore"
+	"github.com/mrz1836/go-datastore"
+	"github.com/newrelic/go-agent/v3/newrelic"
+)
+
+// Validate checks the configuration for specific rules
+func (a *AppConfig) Validate(txn *newrelic.Transaction) error {
+	var err error
+	defer txn.StartSegment("config_validation").End()
+
+	if err = a.Authentication.Validate(); err != nil {
+		return err
+	}
+
+	if err = a.Cachestore.Validate(); err != nil {
+		return err
+	}
+
+	if err = a.Db.Datastore.Validate(); err != nil {
+		return err
+	}
+
+	if err = a.NewRelic.Validate(); err != nil {
+		return err
+	}
+
+	if err = a.Paymail.Validate(); err != nil {
+		return err
+	}
+
+	if err = a.Server.Validate(); err != nil {
+		return err
+	}
+
+	if err = a.validateCachestore(); err != nil {
+		return err
+	}
+
+	if err = a.validateDatastore(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// validateDatastore will check the datastore and validate basic requirements
+func (a *AppConfig) validateDatastore() error {
```

Comment:
Shouldn't be a part of `Datastrore.Validate()` ?

Reply:
Good point, moved